Automattic/wp-calypso
# A4A: Port Payment method form from Jetpack Manage

This PR ports the Payment method form from Jetpack Manage. I have made a few improvements while porting it, converting some of the helper functions to hooks.

<img width="1260" alt="Screenshot 2024-03-13 at 7 57 51 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/fb81e889-ff2e-478e-993b-d9acccdbac91">

Closes https://github.com/Automattic/jetpack-genesis/issues/280

## Proposed Changes

* Port the Payment method form from Jetpack Manage. This includes the following.
     * Credit card fields related components.
     * `getStripeConfiguration` helper function.
     * `useAssignNewCardProcessor` hook. (previously was assignNewCardProcessor helper function)
     * `useSaveCard` hook (previously was saveCreditCard helper function)
     * `useCreateStoredCreditCardMethod` hook.
     * `useReturnUrl` hook.
 * Moved all hooks and lib within the Payment method section folder. This was to make it easier to navigate the dependencies of the Payment method section.
 * **ADDITIONAL**: defined the accent color to update the buttons and fields to align with the A4A branding color. This should also update the rest of the pages.

## Testing Instructions

* Switch branch: `git checkout port/a4a-payment-method-form`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/purchases/payment-methods/add
* Confirm that the form is working as expected and behaves the same way as the one from https://cloud.jetpack.com/partner-portal/payment-methods/add
   **Note that the Submit button does not work at the moment.**


## Pre-merge Checklist
